### PR TITLE
feat(model/linear.py): support norm head for model internlm2

### PR DESCRIPTION
**Pull request by huangting4201**
## Motivation

Add class `InternLM2ScaleColumnParallelLinear` to support norm head for model internlm2.

### CODE REVIEW

File: internlm/model/linear.py [hunk 1]
```diff
@@ -77,20 +82,65 @@ def forward(self, input, gather_dim=0):  # pylint: disable=W0622
...
+        weight_scale: int = 1,
+        norm_head: bool = False,
+    ) -> None:
+        super().__init__(
+            in_features, out_features, process_group, bias=bias, device=device, dtype=dtype, weight_scale=weight_scale
+        )
+
+        self.norm_head = norm_head
+        if self.norm_head:
+            logger.info("Notice norm head is used.")
```
Comment (gaoyang07) [inline internlm/model/linear.py]: 这里logger info的内容可以规范下

File: internlm/model/linear.py [hunk 2]
```diff
@@ -77,20 +82,65 @@ def forward(self, input, gather_dim=0):  # pylint: disable=W0622
...
+
+    Args:
+        in_features (int): size of each input sample
+        out_features (int): size of each output sample
+        process_group (Optional[torch.distributed.ProcessGroup]): The group of the current device for `parallel_mode`.
+        bias (bool): Whether the bias is needed for linears. True by default. But it is typically set to False
+                    in the config.
+        device (Optional[Union[str, torch.device]]): The device will be used.
+        dtype (Optional[torch.dtype]): The type of data.
+        weight_scale (int): For training stability. 1 by default.
```
Comment (gaoyang07) [inline internlm/model/linear.py:97]: 缺了norm head

File: internlm/model/linear.py (hunk 1 above)
Comment (huangting4201) [reply internlm/model/linear.py]: 已更新[1c14206](https://github.com/InternLM/InternEvo/pull/68/commits/1c14206c838c03fdc318b1939388d4c188929ef7)

File: internlm/model/linear.py (hunk 2 above)
Comment (huangting4201) [reply internlm/model/linear.py:97]: 已更新[1c14206](https://github.com/InternLM/InternEvo/pull/68/commits/1c14206c838c03fdc318b1939388d4c188929ef7)

**Review by gaoyang07 — APPROVED** [review]

**Pull request merged**